### DISCUSSION TITLE
Installation fails on Raspberry Pi OS 11 (bullseye)

### DISCUSSION THREAD

**Discussion by miclass** [jekhokie/raspberry-noaa-v2#472] [Q&A]
Installation fails on a fresh new Raspberry Pi OS 11 (bullseye based):
```
TASK [webserver : install php] **************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "No package matching 'php7.2-fpm' is available"}

PLAY RECAP **********************************************************************************************************************************************************************************************************************************
localhost                  : ok=55   changed=48   unreachable=0    failed=1    skipped=12   rescued=0    ignored=0

Error:   Something failed with the install - please inspect the logs above
```
Apparently Raspberry Pi OS 11 has moved to php7.4-fpm.
Any suggestion?

Thank you!

**Comment by ea4aoj** [comment]
The install script and ansible is old, you should download the image directly. If you add the full repository for php install 7.2, you can also look for a file where it looks for php7.2-fpm and change it to your version of PHP. I thus managed to install it on a Bananapi. And everything works correctly. But if you don't want to complicate it, download the image, burn it to a sd card and you just have to configure it, it already comes with php7.2. Greetings.

**Comment by ea4aoj** [comment]
Possibly some of these from the path raspberry-noaa-v2 / blob / master / ansible / roles / common / tasks / dependencies.yml also give you an error like libjpg9, and it is because the repositories are not the same as yours in that debian version. What I did is modify that list for those that I have that are more modern and it worked, although it cost me work. Luck

**Reply by ea4aoj** [reply]
Anyway, the file where it mentioned php7.2-fpm I don't remember where it was, but I found it and changed it to my version of php, but you'll see how later you will find more errors, even if you correct that one.

**Reply by ea4aoj** [reply]
Here you have it, /raspberry-noaa-v2/blob/master/ansible/roles/webserver/tasks/main.yml change it for your version, but I repeat .. you will find more errors, another one is predict, you will have to install by hand.

**Reply by ea4aoj** [reply]
if you notice, this was done from the beginning for the strech repositories, and that debian is already old. name: php repository gpg key (stretch only)
  become: yes
  apt_key:
    url: "https://packages.sury.org/php/apt.gpg"
    state: present
  when: raspbian_version.stdout == 'stretch'

- name: php repository (stretch only)
  become: yes
  apt_repository:
    repo: deb https://packages.sury.org/php/ stretch main
    filename: php7.list
    state: present
  when: raspbian_version.stdout == 'stretch'

- name: install nginx
  become: yes
  apt:
    update_cache: no
    state: present
    name: nginx

- name: install php
  become: yes
  apt:
    update_cache: no
    state: present
    yam:
      - php7.2-fpm
      - php7.2-mbstring
      - php7.2-sqlite3
I repeat, if you don't want to complicate it, download the image.

**Reply by ea4aoj** [reply]
After much sacrifice, I managed to mount it on a debian buster, the machine is a bananapi m2.

![imagen](https://user-images.githubusercontent.com/29492233/141661787-dd11f3ed-c8d0-4fe8-b9ff-2efcb724015a.png)

**Reply by miclass** [reply]
Thank you for your feedbacks...
So if I well understood the installation script is definitively broken for versions after Stretch and even if I would go for Raspberry Pi OS 10 (Buster) I will face errors and dependencies to be corrected manually... 

I am not so happy about using the image, I wuold have prefered to keep control on the OS (updates, other things the Pi cuold do...), anyway I will give a try. 

Just a note: I wuold put crystal clear in the Git homepage these information about the installation scripts, I have spent some time trying.

Thanks!

**Reply by ea4aoj** [reply]
I recommend that you run this script, in theory it installs the missing dependencies.

https://github.com/reynico/raspberry-noaa/blob/master/install.sh

**Reply by miclass** [reply]
This script still looks for the dependencies that cannot be satisfied by Bullseye, it's perhaps even older than the one for v2.
I see a lot of effort participation around this project, I don't understand why the install script is not maintained too...I expect it to be a core part...

**Reply by dom-robinson** [reply]
> This script still looks for the dependencies that cannot be satisfied by Bullseye, it's perhaps even older than the one for v2. I see a lot of effort participation around this project, I don't understand why the install script is not maintained too...I expect it to be a core part...

The original developer @jekhokie is a wizard with ansible and the install setup, but unfortunately life events took over and he has been unable to continue much hands on with the project. The rest of us do the best we can, but I for one (!) am no expert with ansible and the overall install automation. I can follow merge requests and so on, but Im far from able to code 'best practice' :) (Im more of an 'end to end workflow' guy!)  There are others here with some expertise who always do amazing things, but if you guys can see ammendments we should include please please do show us the way and contribute PRs to merge in to improve the platform! 

Thanks folks! Its great to see the interest in the project!!

**Comment by airlomba** [comment]
Hello miclas,

I recently installed raspberry-noaa-v2 in a fresh installation of Bulleye on a Raspberry Pi 4 with latest updates. Also had some hickups when I tried to install it on a different user instead of 'pi'. Once I figured this out, the installation was OK. install_and_upgrade.sh too an eternity (about an hour) due to several missing dependencies, but worked OK by itself.

FYI uname -a returned the following sentence in my system:
Linux pi9 5.10.63-v7l+ #1488 SMP Thu Nov 18 16:15:28 GMT 2021 armv7l GNU/Linux

Best regards,
Emmanuel
(CT7AFR)

**Reply by miclass** [reply]
Thank you Emmanuel!
Currently I am installing a new QFH antenna, once finished I will go back to the raspberry-noaa and try again to let it work on Bullseye!
Kind regards,
-miclass
